tools/integration_tests/gzip/gzip_test.go

```diff
@@ -0,0 +1,173 @@
+		keepCacheControlNoTransform bool // if true, no-transform is reset as ''
+		enableGzipEncodedContent    bool // if true, original file content is gzip-encoded
+		enableGzipContentEncoding   bool // if true, the content is uploaded as gsutil cp -Z i.e. with content-encoding: gzip header in GCS
+	}{
+		{
+			filename:                    textContentWithContentEncodingWithNoTransformFilename,
+			filesize:                    textContentSize,
+			keepCacheControlNoTransform: true,
+			enableGzipEncodedContent:    false,
+			enableGzipContentEncoding:   true,
+		},
+		{
+			filename:                    textContentWithContentEncodingWithoutNoTransformFilename,
+			filesize:                    textContentSize,
+			keepCacheControlNoTransform: false,
+			enableGzipEncodedContent:    false,
+			enableGzipContentEncoding:   true,
+		},
+		{
+			filename:                    gzipContentWithoutContentEncodingFilename,
+			filesize:                    textContentSize,
+			keepCacheControlNoTransform: true, // it's a don't care in this case
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   false,
+		}, {
+			filename:                    gzipContentWithContentEncodingWithNoTransformFilename,
+			filesize:                    textContentSize,
+			keepCacheControlNoTransform: true,
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   true,
+		}, {
+			filename:                    gzipContentWithContentEncodingWithoutNoTransformFilename,
+			filesize:                    textContentSize,
+			keepCacheControlNoTransform: false,
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   true,
+		},
+	}
+
+	for _, fmd := range fmds {
+		var localFilePath string
+		localFilePath, err := helpers.CreateLocalTempFile(fmd.filesize, fmd.enableGzipEncodedContent)
+		if err != nil {
+			return err
+		}
+
+		defer os.Remove(localFilePath)
+
+		// upload to the test-bucket for testing
+		gcsObjectPath := path.Join(setup.TestBucket(), gzipTestResPath, fmd.filename)
+
+		err = helpers.UploadGcsObject(localFilePath, gcsObjectPath, fmd.enableGzipContentEncoding)
+		if err != nil {
+			return err
+		}
+
+		gcsObjectsToBeDeletedEventually = append(gcsObjectsToBeDeletedEventually, gcsObjectPath)
+
+		if !fmd.keepCacheControlNoTransform {
+			err = helpers.ClearCacheControlOnGcsObject(gcsObjectPath)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func destroy_testdata(m *testing.M) error {
+	for _, gcsObjectPath := range gcsObjectsToBeDeletedEventually {
+		err := helpers.DeleteGcsObject(gcsObjectPath)
+		if err != nil {
+			return fmt.Errorf("Failed to delete gcs object gs://%s", gcsObjectPath)
+		}
+	}
+
+	return nil
+}
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	commonFlags := "--sequential-read-size-mb=" + fmt.Sprint(seqReadSizeMb) //+ " --implicit-dirs"
+	flags := [][]string{{commonFlags}}
+	// flags := [][]string{{commonFlags}, {commonFlags, "--enable-storage-client-library=false"}}
```

Comment:
The `--enable-storage-client-library=false` flag is causing mount failures in integration tests.. @Tulsishah any idea ?

Reply:
What exact error you are facing ?

Reply:
@Tulsishah ^

Reply:
Followed up in later PR In https://github.com/GoogleCloudPlatform/gcsfuse/pull/1256/commits/3ac6df6dc005e94f3a2c34964d7d76747bb60f03#r1282723808 .